addons/website/models/ir_ui_view.py

```diff
@@ -305,11 +305,11 @@ def _get_inheriting_views_arch_domain(self, model):
         return expression.AND([website_views_domain, domain])
 
     @api.model
-    def get_inheriting_views_arch(self, model):
+    def _get_inheriting_hierarchy(self):
         if not self._context.get('website_id'):
-            return super(View, self).get_inheriting_views_arch(model)
+            return super(View, self)._get_inheriting_hierarchy()
 
-        views = super(View, self.with_context(active_test=False)).get_inheriting_views_arch(model)
+        views = super(View, self.with_context(active_test=False))._get_inheriting_hierarchy()
         # prefer inactive website-specific views over active generic ones
         return views.filter_duplicate().filtered('active')
```

Comment:
This is not correct: the code expects a recordset, but the super-call returns a dict instead.
I think it would be easier to stick to the former convention, i.e., return views as a recordset.
This makes it easier for overridings.
Building the `hierarchy` tree in `_get_combined_arch()` instead is not an issue, IMHO.